backend/crud.py

```diff
@@ -0,0 +1,211 @@
+        
+        exists = db.execute(
+            select(AttributeDefinition)
+            .where(AttributeDefinition.attribute_id == a.id)
+            .where(AttributeDefinition.schema_id == schema_id)
+        ).scalar()
+        if exists:
+            raise Exception(f'There is already attribute defined')
+
+        ad = AttributeDefinition(
+            attribute_id=attr.attr_id, 
+            schema_id=sch.id,
+            required=attr.required, 
+            list=attr.list, 
+            unique=attr.unique if not attr.list else False,
+            key=attr.key,
+            description=attr.description
+        )
+        db.add(ad)
+        db.flush()
+        if a.type == AttrType.fk:
+            if attr.bind_to_schema is None:
+                raise Exception('You need to bind foreign key to some schema')
+            if attr.bind_to_schema == -1:
+                s = sch
+            else:
+                s = db.execute(select(Schema).where(Schema.id == attr.bind_to_schema)).scalar()
+            if s is None:
+                raise Exception(f'Cannot bind to nonexistent schema with id {attr.bind_to_schema}')
+            bfk = BoundFK(attr_def_id=ad.id, schema_id=s.id)
+            db.add(bfk)
+    db.commit()
+    return sch
+        
+        
+def create_entity(db: Session, schema_id: int, data: dict) -> Entity:
+    sch: Schema = db.execute(select(Schema).where(Schema.id == schema_id)).scalar()
+    if sch is None:
+        raise Exception(f'Schema with id {schema_id} does not exist')
+    e = Entity(schema_id=schema_id)
+    db.add(e)
+    db.flush()
+
+    for field, value in data.items():
+        attr: Attribute = db.execute(select(Attribute).where(Attribute.name == field)).scalar()
+        if attr is None:
+            raise Exception(f'Attribute with name `{field}` does not exist')
+
+        attr_def: AttributeDefinition = db.execute(
+            select(AttributeDefinition)
+            .where(AttributeDefinition.schema_id == schema_id)
+            .where(AttributeDefinition.attribute_id == attr.id)
+        ).scalar()
+        if attr_def is None:
+            raise Exception(f'There is no attribute definition for schema id {schema_id} and attribute id {attr.id}')
+
+        model: Value = val_map.get(AttrType(attr.type))
+        if model is None:
+            raise Exception(f'No value model found for type `{attr.type}`')
+        
+        caster = type_map.get(AttrType(attr.type))
+        if caster is None:
+            raise Exception(f'No caster found for type `{attr.type}`')
+
+        try:
+            if isinstance(value, list):
+                if not attr_def.list:
+                    raise Exception(f'Attribute {attr.name} on schema {sch.name} cannot hold multiple values')
+                values = [caster(i) for i in value]
+            else:
+                values = [caster(value)]
+        except ValueError as e:
+            raise # can this even happen with validation done before?
```

Comment:
I assume that you want to use a dynamically created pydantic model for validation. If my understanding is correct, this should also take care of the type casting. If that is the case, this explicit type casting and exception handling might become obsolete.

Reply:
> this should also take care of the type casting
Yeah, that's what was on my mind. Well, let's leave it they way it is until there is code for creating dynamic schemas😉